cpp/util/SnowflakeCommon.cpp

```diff
@@ -119,5 +122,19 @@ CURLcode set_curl_proxy(CURL *curl, const char* proxy, const char* no_proxy)
   }
 }
 
+void STDCALL sf_exception_on_memory_failure()
+{
+  // should be called by odbc driver snowflake_global_init
+  exception_on_memory_error = true;
+}
+
+void STDCALL sf_memory_error_handler()
+{
+  if (exception_on_memory_error)
+    throw std::bad_alloc();
```

Comment:
Please make odbc side change and post your branch as well. I'd like to see the changes there to ensure this throw can be handled easily there and the relative change there is minor.

Reply:
Checked implementation on ODBC side currently it would never trigger the functions defined in memory.c.
Will call sf_memory_error_handler() on ODBC side for possible further extension but for now no visible impact with that.